src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package cinema.dao.impl;
+
+import cinema.dao.ShoppingCartDao;
+import cinema.exceptions.DataProcessingException;
+import cinema.lib.Dao;
+import cinema.model.ShoppingCart;
+import cinema.model.User;
+import cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert shoppingCart Entity "
+                    + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> getByUserQuery = session.createQuery("FROM ShoppingCart "
+                    + "WHERE user.id = :user_id", ShoppingCart.class);
```

Comment:
will you have access to user of this shoppingCart after session has been closed?

Reply:
done